### DISCUSSION TITLE
Where to find Types

### DISCUSSION THREAD

**Discussion by jmhacks** [wevm/wagmi#171] [Q&A]
Hi! I'm using WAGMI with Typescript. Wondering if there are types I can import to use throughout my app related to the different return objects from hooks? I've just been defining them on my own right now. Thanks:)

**Comment by rcstanciu** [comment]
Hi! Which hooks in particular are you using and don't have types? Almost everything should be typed, except `useContract`. Although, the `useContract` can be easily typed using, for example, https://github.com/dethcrypto/TypeChain
